### DISCUSSION TITLE
Pygame issue

### DISCUSSION THREAD

**Discussion by Laenpo** [community/community#46801] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi, ONce I install pygame, the code occur below
pip install pygame
Collecting pygame
Using cached pygame-2.1.2.tar.gz (10.1 MB)
Preparing metadata (setup.py) ... error
error: subprocess-exited-with-error

× python setup.py egg_info did not run successfully.
│ exit code: 1
note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.

When I download pygame on mac it's always like this and I have no idea what's the problem.

**Comment by mammaddrik** [comment, marked as answer]
## Install Pygame in MacOS

At first, open the Terminal which is located at Applications -> Utilities -> Terminal and make sure that the Python3 is installed in your system. If it is not installed then type the below command.
`$ brew install python3`


## Installing dependency
XCode: The first step for PyGame is to install Apple’s Xcode program. In your Terminal app, enter and run the following command to install XCode.
`$ xcode-select --install`
Be sure to click on all the verification recommendations that require XCode.

## Installing PyGame
We can now install the latest version of PyGame. Type the following command in Terminal and press Enter.
`$ pip3 install pygame`
or
`$ pip install pygame`


To test if PyGame has been installed on your Mac, open terminal and type python, and import pygame as follows:
`python`
`import pygame`

If you don’t see any errors, it means that PyGame has been successfully installed on your Mac.

bobbyhadz.com [Here](https://bobbyhadz.com/blog/python-error-metadata-generation-failed-encountered-error) 
stackoverflow.com [Here](https://stackoverflow.com/questions/70916814/error-metadata-generation-failed-cant-install-artic-module)

I hope I was able to solve your problem.

**Reply by Laenpo** [reply]
Hi, I had already downloaded xcode and homebrew before. However, the problem still exists

**Reply by mammaddrik** [reply]
According to the explanation in the error text, the problem is not from pip and the timed out occurred while accessing the resources, the problem can probably be solved with VPN.  Please test.

**Reply by Laenpo** [reply]
Hi, the problem was resolved by dowloading homebrew and update it. Thank you for the answer !
